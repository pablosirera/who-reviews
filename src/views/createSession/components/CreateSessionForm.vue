<template>
  <form @submit.prevent="onSubmit" class="flex flex-col w-48 my-0 mx-auto">
    <input
      v-model="name"
      type="text"
      name="name"
      data-testid="session-name"
      class="border outline-none py-2 px-4 rounded"
      :placeholder="$t('create.name')"
    />
    <div class="mt-2 cursor-pointer" @click="toggleIsActive()">
      <input v-model="isActive" type="checkbox" name="isActive" />
      <span class="mx-2">{{ $t('create.isActive') }}</span>
    </div>
    <div class="flex justify-center">
      <TheButton :text="$t('create.create')" size="small" />
    </div>
  </form>
</template>

<script>
import TheButton from '@/components/ui/TheButton'

export default {
  name: 'CreateSessionForm',
  components: {
    TheButton,
  },
  data: () => ({
    name: '',
    isActive: true,
  }),
  methods: {
    onSubmit() {
      this.$emit('submit-form', { name: this.name, active: this.isActive })
    },
    toggleIsActive() {
      this.isActive = !this.isActive
    },
  },
}
</script>
