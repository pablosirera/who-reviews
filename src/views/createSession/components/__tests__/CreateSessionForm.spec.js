import { shallowMount } from '@vue/test-utils'
import CreateSessionForm from '../CreateSessionForm.vue'

describe('CreateSession -> CreateSessionForm', () => {
  const build = () => {
    const wrapper = shallowMount(CreateSessionForm)

    return {
      wrapper,
    }
  }

  it('should match snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit event if form is valid', () => {
    const { wrapper } = build()

    const input = wrapper.find('[data-testid="session-name"]')

    input.setValue('name of session')
    wrapper.trigger('submit')

    const formSubmittedCalls = wrapper.emitted('submit-form')
    expect(formSubmittedCalls).toHaveLength(1)
  })
})
