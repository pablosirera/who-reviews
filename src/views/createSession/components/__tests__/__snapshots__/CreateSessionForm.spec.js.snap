// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateSession -> CreateSessionForm should match snapshot 1`] = `
<form
  class="flex flex-col w-48 my-0 mx-auto"
>
  <input
    class="border outline-none py-2 px-4 rounded"
    data-testid="session-name"
    name="name"
    placeholder="create.name"
    type="text"
  />
   
  <div
    class="flex justify-center"
  >
    <thebutton-stub
      size="small"
      text="create.create"
    />
  </div>
</form>
`;
